lib/puppet-strings/tasks/gh_pages.rb

```diff
@@ -33,16 +31,19 @@
       end
     end
 
-    desc 'Push new docs to GitHub.'
     task :push do
+      output = `git describe --long 2>/dev/null`
+      # If a project has never been tagged, fall back to latest SHA
+      output.empty? ? git_sha = `git log --pretty=format:'%H' -n 1` : git_sha = output
+
       Dir.chdir('doc') do
         system 'git add .'
-        system "git commit -m '[strings] Generated Documentation Update'"
+        system "git commit -m '[strings] Generated Documentation Update at Revision #{git_sha}'"
         system 'git push origin gh-pages -f'
       end
     end
 
-    desc 'Run checkout, generate, and push tasks.'
+    desc 'Run checkout, generate, configure, and push tasks.'
```

Comment:
One last thing, we should probably make this description just say "update docs and push to github" or something like that, since we aren't advertising the underlying tasks anymore.